Source/WebKit/WebProcess/WebPage/libwpe/AcceleratedSurfaceLibWPE.h

```diff
@@ -41,28 +41,20 @@ class AcceleratedSurfaceLibWPE final : public AcceleratedSurface {
     WTF_MAKE_NONCOPYABLE(AcceleratedSurfaceLibWPE);
     WTF_MAKE_TZONE_ALLOCATED(AcceleratedSurfaceLibWPE);
 public:
-    static std::unique_ptr<AcceleratedSurfaceLibWPE> create(WebPage&, Client&);
+    static std::unique_ptr<AcceleratedSurfaceLibWPE> create(WebPage&, Function<void()>&& frameCompleteHandler);
     ~AcceleratedSurfaceLibWPE();
 
     uint64_t window() const override;
     uint64_t surfaceID() const override;
     void clientResize(const WebCore::IntSize&) override;
-    bool shouldPaintMirrored() const override
-    {
-#if PLATFORM(GTK) && !USE(GTK4)
-        return true;
-#else
-        return false;
-#endif
-    }
-
-    void initialize() override;
     void finalize() override;
     void willRenderFrame() override;
     void didRenderFrame(WebCore::Region&&) override;
 
 private:
-    AcceleratedSurfaceLibWPE(WebPage&, Client&);
+    AcceleratedSurfaceLibWPE(WebPage&, Function<void()>&& frameCompleteHandler);
+
+    void initialize();
```

Comment:
shouldn't this override?

Reply:
No, I removed the virtual initialize from the parent because it's only needed by this class. My initial idea was to remove initialize and move the code to window(), because the caller always does initialize() + window(), but window() is const and we don't want to change that. So, the solution was to keep the initialize() as private non const member that is called from window() with the dirty const cast.